File: task-03/Go/4/Sub-Task4.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func generateDiamond(n int) string {
	var builder strings.Builder

	for i := 0; i < n; i++ {
		builder.WriteString(strings.Repeat(" ", n-i-1))
		builder.WriteString(strings.Repeat("*", 2*i+1))
		builder.WriteString("\n")
	}

	for i := n - 2; i >= 0; i-- {
		builder.WriteString(strings.Repeat(" ", n-i-1))
		builder.WriteString(strings.Repeat("*", 2*i+1))
		builder.WriteString("\n")
	}

	return builder.String()
}

func main() {
	inputFile, err := os.Open("input.txt")
	if err != nil {
		fmt.Println("Error opening input file:", err)
		return
	}
	defer inputFile.Close()

	var n int
	scanner := bufio.NewScanner(inputFile)
	if scanner.Scan() {
		_, err := fmt.Sscanf(scanner.Text(), "%d", &n)
		if err != nil {
			fmt.Println("Error reading the number from input file:", err)
			return
		}
	}

	diamondPattern := generateDiamond(n)

	outputFile, err := os.Create("output.txt")
	if err != nil {
		fmt.Println("Error opening output file:", err)
		return
	}
	defer outputFile.Close()

	_, err = outputFile.WriteString(diamondPattern)
	if err != nil {
		fmt.Println("Error writing to output file:", err)
		return
	}

	fmt.Println("Pattern printed to output.txt")
}

File: task-03/C/4/Sub-Task4.c
#include <stdio.h>
#include <stdlib.h>

void generate_diamond(int n, FILE *output_file) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            fputc(' ', output_file);
        }
        for (int j = 0; j < 2 * i + 1; j++) {
            fputc('*', output_file);
        }
        fputc('\n', output_file);
    }

    for (int i = n - 2; i >= 0; i--) {
        for (int j = 0; j < n - i - 1; j++) {
            fputc(' ', output_file);
        }
        for (int j = 0; j < 2 * i + 1; j++) {
            fputc('*', output_file);
        }
        fputc('\n', output_file);
    }
}

int main() {
    FILE *input_file = fopen("input.txt", "r");
    if (input_file == NULL) {
        perror("Error opening input file");
        return EXIT_FAILURE;
    }

    int n;
    if (fscanf(input_file, "%d", &n) != 1) {
        fprintf(stderr, "Error reading the number from input file.\n");
        fclose(input_file);
        return EXIT_FAILURE;
    }
    fclose(input_file);

    FILE *output_file = fopen("output.txt", "w");
    if (output_file == NULL) {
        perror("Error opening output file");
        return EXIT_FAILURE;
    }

    generate_diamond(n, output_file);

    fclose(output_file);
    printf("Pattern printed to output.txt\n");

    return EXIT_SUCCESS;
}
